### ISSUE TITLE
4.0b4 Crash loading in Snapshot, used GEORAM+RamBATTERY+Drive8

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#632]
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
Assertion failed: (cartridge), function _load, file /Users/hoff/virtualc64/Emulator/Peripherals/ExpansionPort.cpp, line 50.
[Untitled.zip](https://github.com/dirkwhoffmann/virtualc64/files/5906361/Untitled.zip)

**Comment by dirkwhoffmann** [comment]
I've fixed it by cleaning up the expansion port code a bit. I've made `cartridge` a `unique_ptr` (which I wanted to do anyway).

Old:

 ```C++
    // Attached cartridge (nullptr if no cartridge is plugged in)
    Cartridge *cartridge = nullptr;
```

New:

```C++
    // Attached cartridge
    std::unique_ptr<Cartridge> cartridge;
```

**Comment by dirkwhoffmann** [comment]
Should be fixed in v4.0b5.
